client/my-sites/customer-home/main.jsx

```diff
@@ -89,6 +95,19 @@ const Home = ( {
 		}
 	}, [ isSuccess ] );
 
+	useEffect( () => {
+		if ( ! isAtomicSiteLaunching && wasAtomicSiteLaunched ) {
+			queryClient.invalidateQueries( getCacheKey( siteId ) );
+			setWasAtomicSiteLaunched( false );
+		}
+	}, [ isAtomicSiteLaunching, queryClient, siteId, wasAtomicSiteLaunched ] );
+
+	useEffect( () => {
+		if ( isAtomicSiteLaunching ) {
+			setWasAtomicSiteLaunched( true );
+		}
+	}, [ isAtomicSiteLaunching ] );
```

Comment:
We probably need another one of these `useEffect` blocks to reset the `wasAtomicSiteLaunched` back to `false` again.

```ts
useEffect( () => {
   setWasAtomicSiteLaunched( false );
}, [ siteId ] );
```

If you're on My Home and then switch sites using the site switcher then `<Home />` won't be unmounted, the props will just get updated. _Technically_ if the user switched sites in the middle of launching then it'd be the new home layout's query that would get invalidated, not the old one's.

Reply:
Another way to do it would be to make `wasAtomicSiteLaunched` into a site ID maybe? If `wasAtomicSiteLaunched` equals the current `siteId` then you invalidate the query. Otherwise you just ignore it.

Reply:
@p-jackson I went with the `siteId` suggestion, can you give it another run?